odbc/src/api.rs

```diff
@@ -0,0 +1,1047 @@
+#![allow(unused)]
+/// ^ TODO: remove that when implementations are complete
```

Comment:
I have a couple quibbles with this:
- I don't love that this is going to result in us not getting this lint at all until we've completed the entire API. Especially for things like remembering to use all the appropriate out params and length args, I think it is a useful lint for implementing a big C API like this
- Even once we _have_ completed the entire API, there will still be functions that don't use their arguments

As an alternative, I'd suggest that we prefix all of the function args with `_` in this PR. As we implement functions, we can remove the leading `_` from the params we use, and leave all others